module/userProjects/helper.js

```diff
@@ -4550,6 +4490,93 @@ module.exports = class UserProjectsHelper {
 			}
 		})
 	}
+
+	/**
+	 * Update ACL for a project if user owns the project and submission level is ENTITY.
+	 *
+	 * @param {String} projectId - The ID of the project whose ACL should be updated
+	 * @param {Object} bodyData - The request body containing ACL updates
+	 * @param {Object} userDetails - Logged-in user's details object
+	 * @returns {Promise<Object>} - Response object containing update status, message, and updated ACL data
+	 *
+	 * @throws {Object} - Throws an error if submission level is invalid or project update fails
+	 */
+	static updateAcl(projectId, bodyData, userDetails) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Only allow updates if submission level is ENTITY
+				if (process.env.SUBMISSION_LEVEL !== 'ENTITY') {
+					throw {
+						success: false,
+						message: CONSTANTS.apiResponses.SUBMISSION_LEVEL_NOT_COMPLIED,
+					}
+				}
+
+				// Extract user and tenant IDs
+				const userId = userDetails.userInformation.userId
+				const tenantId = userDetails.userInformation.tenantId
+
+				// Check if project exists and belongs to the user
+				const projectData = await projectQueries.projectDocument({
+					_id: projectId,
+					userId,
+					tenantId,
+				})
+
+				if (!projectData || projectData.length === 0) {
+					throw {
+						success: false,
+						message: CONSTANTS.apiResponses.PROJECT_DOES_NOT_BELONG_TO_USER,
+					}
+				}
+
+				// Update ACL field in the project
+				const updatedProject = await projectQueries.findOneAndUpdate(
+					{
+						_id: projectId,
+						userId,
+						tenantId,
+					},
+					{
+						$set: { acl: bodyData.acl },
```

Comment:
@Prajwal17Tunerlabs we should have validations for bodydata.acl , and it can be part of api validator logic